client/post-editor/editor-image-selector/index.jsx

```diff
@@ -0,0 +1,163 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { connect } from 'react-redux';
+import { isEqual, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import EditorImageSelectorPreview from './preview';
+import ImageSelectorDropZone from './dropzone';
+import isDropZoneVisible from 'state/selectors/is-drop-zone-visible';
+import MediaLibrarySelectedData from 'components/data/media-library-selected-data';
+import MediaActions from 'lib/media/actions';
+import MediaModal from 'post-editor/media-modal';
+import MediaStore from 'lib/media/store';
+import { localize } from 'i18n-calypso';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+export class EditorImageSelector extends Component {
+	static propTypes = {
+		hasDropZone: PropTypes.bool,
+		imageIds: PropTypes.array,
+		isDropZoneVisible: PropTypes.bool,
+		showEditIcon: PropTypes.bool,
+		maxWidth: PropTypes.number,
+		selecting: PropTypes.bool,
+		translate: PropTypes.func,
+		onImageChange: PropTypes.func,
+		onImageSelected: PropTypes.func,
+		onRemoveImage: PropTypes.func,
+	};
+
+	static defaultProps = {
+		imageIds: [],
+		hasDropZone: false,
+		isDropZoneVisible: false,
+		maxWidth: 450,
+		onImageSelected: noop,
```

Comment:
instead of having 3 distinct functions for selected, change, remove - why not just have one onchange event?

Reply:
Disregard - I see it is the underlying components that declare these :)